### PR TITLE
 [FFM-9560] - Refactor and cleanup code

### PR DESCRIPTION
**Pull request by andybharness**
[FFM-9560] - Refactor and cleanup code

    What
    - Break down large methods
    - Remove unused code
    - Remove boiler plate
    - Remove unneeded dependencies
    - Remove compiler warnings
    - Lint messages
    - Sonarqube warnings
    - etc

    Why
    We want to keep the SDK code as maintainable as possible and reduce clutter

    Testing
    Manual + unit

[FFM-9560]: https://harness.atlassian.net/browse/FFM-9560?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: cfsdk/src/main/java/io/harness/cfsdk/cloud/analytics/AnalyticsPublisherService.java [hunk 1]
```diff
@@ -30,42 +29,25 @@ public class AnalyticsPublisherService {
...
-    }
+    private static final String CLIENT = "client";
+    private static final String SDK_TYPE = "SDK_TYPE";
+    private static final String SDK_VERSION = "SDK_VERSION";
+    private static final String SDK_LANGUAGE = "SDK_LANGUAGE";
+    private static final String TARGET_ATTRIBUTE = "target";
+    private static final String FEATURE_IDENTIFIER_ATTRIBUTE = "featureIdentifier";
+    private static final String FEATURE_NAME_ATTRIBUTE = "featureName";
+    private static final String VARIATION_IDENTIFIER_ATTRIBUTE = "variationIdentifier";
+    private static final String VARIATION_VALUE_ATTRIBUTE = "variationValue";
```
Comment (erdirowlands) [inline cfsdk/src/main/java/io/harness/cfsdk/cloud/analytics/AnalyticsPublisherService.java:40]: No big deal but `variationValue` isn't required in the evaluation metrics payload. https://harness.atlassian.net/wiki/spaces/FFM/pages/21070348431/Feature+Flag+SDKs#Metric-Attributes

Comment (erdirowlands) [reply cfsdk/src/main/java/io/harness/cfsdk/cloud/analytics/AnalyticsPublisherService.java:40]: Might not be worth the effort to remove it. Up to you

File: cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java [hunk 2]
```diff
@@ -10,12 +10,7 @@ public class Target {
 
     private String name;
     private String identifier;
-    private final Map<String, Object> attributes;
-
-    {
-
-        attributes = new HashMap<>();
-    }
+    private final Map<String, Object> attributes = new HashMap<>();
```
Comment (erdirowlands) [inline cfsdk/src/main/java/io/harness/cfsdk/cloud/model/Target.java:13]: Can't quite believe the previous style here. Nice one.

File: cfsdk/src/main/java/io/harness/cfsdk/CfClient.java [hunk 3]
```diff
@@ -873,43 +379,16 @@ public JSONObject jsonVariation(String evaluationId, JSONObject defaultValue) {
...
-    }
-
     /**
      * Clears the occupied resources and shut's down the sdk.
      * After calling this method, the {@link #initialize} must be called again. It will also
      * remove any registered event listeners.
      */
     @Override
-    public void destroy() {
+    public void close() {
```
Comment (erdirowlands) [inline cfsdk/src/main/java/io/harness/cfsdk/CfClient.java:388]: Definitely a better name. Forgive me if this is a stupid question, but if we change the public method name here, will that not break backwards compatability?

Comment (erdirowlands) [reply cfsdk/src/main/java/io/harness/cfsdk/CfClient.java:388]: Oh nevermind. See the deprecated for `destroy` below! Ignore.

**Review by erdirowlands — APPROVED** [review]

File: cfsdk/src/main/java/io/harness/cfsdk/CfClient.java (hunk 3 above)
Comment (andybharness) [reply cfsdk/src/main/java/io/harness/cfsdk/CfClient.java:388]: yeah the main problem here is destroy() is non-standard and can't be used in try-with-resources statements. it was re-added it back in as deprecated to allow developers time to upgrade. then we can remove it later

**Pull request merged**